api/v1alpha1/api.go

```diff
@@ -205,3 +211,117 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// BackendSecurityPolicyType specifies the type of auth mechanism used to access a backend.
+type BackendSecurityPolicyType string
+
+const (
+	BackendSecurityPolicyTypeAPIKey BackendSecurityPolicyType = "APIKey"
+	BackendSecurityPolicyTypeAWSIAM BackendSecurityPolicyType = "AWS_IAM"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication rules on access the provider from the Gateway.
+//
+// Only one type of BackendSecurityPolicy can be defined.
+// +kubebuilder:validation:MaxProperties=2
+type BackendSecurityPolicySpec struct {
+	// Type specifies the auth mechanism used to access the provider. Currently, only "APIKey", AND "AWS_IAM" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;AWS_IAM
+	Type BackendSecurityPolicyType `json:"type"`
+
+	// APIKey is a mechanism to access a backend(s). The API key will be injected into the Authorization header.
+	//
+	// +optional
+	APIKey *BackendSecurityPolicyAPIKey `json:"apiKey,omitempty"`
+
+	// CloudProviderCredentials is a mechanism to access a backend(s). Cloud provider specific logic will be applied.
+	//
+	// +optional
```

Comment:
could you add comment about AWS_IAM will be interpreted to this field?

Reply:
well, shouldn't the type be "CloudProviders" instead of AWS_IAM? The "type" field is for union type assertion and this feels a weird because AWS_IAM is not for the union type assertion but for the higher layer. @yuzisun @arkodg